analyzer/src/main/kotlin/managers/utils/DependencyHandler.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.managers.utils
+
+import org.ossreviewtoolkit.analyzer.PackageManager
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtIssue
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+
+/**
+ * An interface used by [DependencyGraphBuilder] to handle the specific representations of concrete [PackageManager]
+ * implementations in a generic way.
+ *
+ * A package manager may use its own, internal representation of a dependency. When constructing the
+ * [org.ossreviewtoolkit.model.DependencyGraph] by passing the dependencies of the single scopes, the builder must be
+ * able to extract certain information from the dependency objects. This is done via an implementation of this
+ * interface.
+ *
+ * @param D the type of the dependency representation
```

Comment:
Nit: Could you find a way to include the docs for `D` into the fluent text, to completely avoid "Javadoc"-style docs?

Reply:
Done. Included a reference in the comment text.